### PR TITLE
feat(stackable-versioned): Add K8s specific features

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/issues/issues/507

```rust
#[versioned(
    version(name = "v1alpha1"),
    version(name = "v1beta1"),
    version(name = "v1"),
    # 👇 new code
    k8s(group = "stackable.tech")
)]
#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
pub struct FooSpec {
    #[versioned(
        added(since = "v1beta1"),
        changed(since = "v1", from_name = "bah", from_type = "u16")
    )]
    bar: usize,
    baz: bool,
}
```

You can also generate merged CRDs:

```rust
let crd = Foo::merged_crd("v1")?;
let yaml = serde_yaml::to_string(&merged_crd)?;
```

### CODE REVIEW

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

File: crates/stackable-versioned-macros/tests/crd.rs
[the remark this answers is not in the document]
Comment (Techassi) [reply crates/stackable-versioned-macros/tests/crd.rs:1]: Done in 06b9106

File: crates/stackable-versioned-macros/tests/trybuild.rs
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/tests/trybuild.rs:1]: The commented out mod definitions will break without wrapping them, eg:

```
mod default {
  mod fail {
    ...
  }

  mod fail {
    ...
  }
}

mod k8s {
  ...
}
```

Comment (Techassi) [reply crates/stackable-versioned-macros/tests/trybuild.rs:1]: Addressed in 2a3ba10

File: crates/stackable-versioned-macros/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,9 @@ license.workspace = true
 edition.workspace = true
 repository.workspace = true
 
+[package.metadata.cargo-udeps.ignore]
+development = ["schemars", "serde_yaml"]
+
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/Cargo.toml:11]: Can we add an explanation why they are ignored, for posterity?
Will there be a time they can be removed from the list?

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM, one command, non-blocker

File: crates/stackable-versioned-macros/Cargo.toml (hunk 1 above)
Comment (Techassi) [reply crates/stackable-versioned-macros/Cargo.toml:11]: Ah yes, I can add such a comment.

> Will there be a time they can be removed from the list?

They can eventually be removed once cargo-udeps supports checking dev-dependencies (which are used in tests).

Comment (Techassi) [reply crates/stackable-versioned-macros/Cargo.toml:15]: Added in 9529a50

File: crates/stackable-versioned-macros/Cargo.toml [hunk 2]
```diff
@@ -6,6 +6,10 @@ license.workspace = true
 edition.workspace = true
 repository.workspace = true
 
+# cargo-udeps throws an error that these dependencies are unused. They are,
+# however, used in K8s specific test cases. This is a false-positive and a known
+# limitation of cargo-udeps. These entries can be removed once cargo-udeps
+# supports detecting usage of such dependencies.
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned-macros/Cargo.toml:12]:
```suggestion
# cargo-udeps throws an error that these dependencies are unused. They are,
# however, used in K8s specific test cases. This is a false-positive and an
# apparent limitation of cargo-udeps. These entries can be removed once 
# cargo-udeps supports detecting usage of such dependencies.
```

**Review by NickLarsenNZ — APPROVED** [review]

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**